src/system_identity/iam_client.rs

```diff
@@ -0,0 +1,189 @@
+use base64::{engine::general_purpose, Engine};
+use http::{
+    header::{AUTHORIZATION, CONTENT_TYPE},
+    StatusCode, Uri,
+};
+use serde::{Deserialize, Deserializer};
+use serde_json::json;
+use thiserror::Error;
+
+use crate::{
+    http_client::{HttpClient, HttpClientError},
+    token::AccessToken,
+};
+
+#[derive(Debug, Clone, Error)]
+pub enum IAMClientError {
+    #[error("error creating system identity: `{0}`")]
+    IAMClientError(String),
+}
+
+pub trait IAMClient {
+    fn create_system_identity(
+        &self,
+        token: &AccessToken,
+        pub_key: &[u8],
+    ) -> Result<SystemIdentityCreationResponseData, IAMClientError>;
+}
+
+/// Implementation of the IAMClient trait for a generic HTTP client.
+pub struct HttpIAMClientImpl<'a, C: HttpClient> {
```

Comment:
```suggestion
pub struct HttpIAMClient<'a, C: HttpClient> {
```

Reply:
Done in https://github.com/newrelic/newrelic-auth-rs/pull/25/commits/709b78c40c982efd198af8ac77f457bc80040c9d!